test/cppsuite/src/main/test.cpp

```diff
@@ -135,12 +142,16 @@ test::run()
       db_create_config, _args.home.empty() ? DEFAULT_DIR : _args.home);
 
     /* Initiate the load stage of each component. */
-    for (const auto &it : _components)
-        it->load();
+    for (const auto &it : _components) {
+        if (it->enabled())
+            it->load();
+    }
 
     /* Spawn threads for all component::run() functions. */
-    for (const auto &it : _components)
-        _thread_manager->add_thread(&component::run, it);
+    for (const auto &it : _components) {
```

Comment:
Why are these separate loops?

Reply:
We want to load all the components first then we can run all of them.